### PR TITLE
Deleted legacy pip_import rule

### PR DESCRIPTION
**Pull request by UebelAndre**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](https://github.com/bazelbuild/rules_python/blob/master/CONTRIBUTING.md) for info
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

As described in https://github.com/bazelbuild/rules_python/issues/436#issuecomment-985699038, maybe it's time to clean out the old pip repository rules. In investigating cache invalidation issues. I've often been confused by what code path was being used since the two are quite similar.

Issue Number: https://github.com/bazelbuild/rules_python/issues/436#issuecomment-985699038

## What is the new behavior?
This change deletes some old rules that have long since been considered legacy and have modern replacements that are actively being developed.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by UebelAndre** [comment]
@alexeagle @hrfuller Based on the conversation at https://github.com/bazelbuild/rules_python/issues/436#issuecomment-985699038, I've opened this PR. I think the rules have been communicated as EOL for some time so just deleting them feels acceptable to me but if you're hoping for a more transitional closure this code then I'm happy to defer to a PR from either one of you 😄

File: .gitignore
Comment (hrfuller) [inline .gitignore:40]: Is this a drive-by?

File: examples/pip_install/.bazelrc
Comment (hrfuller) [inline examples/pip_install/.bazelrc:2]: This change seems good but also unrelated to this PR. Should it be its own?

**Review by hrfuller — APPROVED** [review]
LGTM, modulo comment about PR hygiene.

File: examples/pip_install/.bazelrc
Comment (UebelAndre) [reply examples/pip_install/.bazelrc:2]: Does it matter hugely if this is included? I was a way for me to test the changes since the examples are already a little difficult to run. I found this to be necessary in validating the changes were safe and would opt to leave it in.

File: .gitignore
Comment (UebelAndre) [reply .gitignore:40]: This is a recommended pattern and was used in testing this Pr. I'd recommend it stay if possible

**Comment by UebelAndre** [comment]
@hrfuller i think the changes here should be merged as is. Do you feel super strongly otherwise?

File: tools/BUILD
Comment (alexeagle) [inline tools/BUILD:29]: I bet there's also some docs or checks for these files we can remove

**Review by alexeagle — APPROVED** [review]
Thanks!

File: tools/BUILD
Comment (UebelAndre) [reply tools/BUILD:29]: You were right. I found the CONTRIBUTING docs and removed them.

**Review by alexeagle — APPROVED** [review]
ooh this also has the amazing side-effect of reducing the size of this repo enough that I think we can just ship the GH archive file rather than uploading our own release artifact

File: examples/pip_install/.bazelrc
Comment (hrfuller) [reply examples/pip_install/.bazelrc:2]: It doesn't matter hugely. In cases where we have to revert functionality it can matter, because this is a feature we want regardless of this PR.

**Pull request merged**
